CurrencyConverter/CurrencyConverter.WebApi/Middleware/ErrorHandling/ErrorHandlingMiddleware.cs

```diff
@@ -0,0 +1,76 @@
+﻿using CurrencyConverter.Contracts;
+using System;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.Hosting;
+using Newtonsoft.Json;
+using Serilog;
+
+namespace CurrencyConverter.WebApi.Middleware.ErrorHandling
+{
+    public class ErrorHandlingMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger _logger;
+        private readonly IWebHostEnvironment _environment;
+
+        public ErrorHandlingMiddleware(RequestDelegate next, ILogger logger, IWebHostEnvironment environment)
+        {
+            _next = next;
+            _logger = logger;
+            _environment = environment;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (Exception ex)
+            {
+                await LogErrorExceptionWithRequestBody(context, ex);
+                await HandleExceptionAsync(context, ex);
+            }
+        }
+
+        private Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            context.Response.ContentType = "application/json";
+            context.Response.StatusCode = (int)HttpStatusCode.InternalServerError;
+
+            var errorMessage = "Internal server error happened. Please contact support";
```

Comment:
Where is this outputted? Since as I understand the exception with it's stuff is logged in log file insider server?

Reply:
It's logged to the file, and also the user of API would get this message as a response as well ( and if it's in development mode, the error code with it as well)